hazelcast/serialization/service.py

```diff
@@ -78,24 +78,40 @@ def _register_constant_serializers(self):
         self._registry.safe_register_serializer(self._registry._python_serializer)
 
     def register_class_definitions(self, class_definitions, check_error):
-        class_defs = dict()
+        factories = dict()
         for cd in class_definitions:
-            if cd in class_defs:
-                raise HazelcastSerializationError("Duplicate registration found for class-id: %s" % cd.class_id)
-            class_defs[cd.class_id] = cd
+            factory_id = cd.factory_id
+            class_defs = factories.get(factory_id, None)
+            if class_defs is None:
+                class_defs = dict()
+                factories[factory_id] = class_defs
+
+            class_id = cd.class_id
+            if class_id in class_defs:
+                raise HazelcastSerializationError("Duplicate registration found for class-id: %s" % class_id)
+            class_defs[class_id] = cd
+
         for cd in class_definitions:
-            self.register_class_definition(cd, class_defs, check_error)
+            self.register_class_definition(cd, factories, check_error)
 
-    def register_class_definition(self, cd, class_defs, check_error):
+    def register_class_definition(self, cd, factories, check_error):
```

Comment:
nit: it may be a good idea to rename `cd` into `class_definition` to keep the same level of verbosity for parameter names as in `register_class_definitions(self, class_definitions, check_error)`

Reply:
Renamed it